core/src/main/kotlin/io/customer/sdk/communication/EventBus.kt

```diff
@@ -55,4 +61,17 @@ class EventBusImpl(
         jobs.forEach { it.cancel() }
         jobs.clear()
     }
+
+    override fun <T : Event> subscribe(type: KClass<T>, action: suspend (T) -> Unit): Job {
+        val job = scope.launch {
+            flow.filter { type.isInstance(it) }.mapNotNull { it.safeCast(type) }.collect { event ->
```

Comment:
Feels like unnecessary looping. Can we do something like this?

```suggestion
            flow.mapNotNull { it.safeCast(type) }.collect { event ->
```

Reply:
good catch, it was left from when I was force casting 😄